File: c/pangram/src/pangram.c
#include <string.h>
#include <ctype.h>

#include "pangram.h"

#define chars_len ('z' - 'a' + 1)

bool is_pangram(const char *sentence) {
    bool chars[chars_len] = {0};
    int len;
    if (sentence == NULL || (len = strlen(sentence)) == 0) {
        return false;
    }
    for (int i = 0; i < len; i++) {
        if (isalpha(sentence[i])) {
            chars[tolower(sentence[i]) - 'a'] = true;
        }
    }
    for (int i = 0; i < chars_len; i++) {
        if (!chars[i]) {
            return false;
        }
    } 
    return true;
}

File: go/pangram/pangram.go
package pangram

import "strings"

func IsPangram(input string) bool {
	c := make([]bool, 'z'-'a'+1)
	for _, r := range strings.ToLower(input) {
		if r >= 'a' && r <= 'z' {
			c['z'-r] = true
		}
	}
	for _, v := range c {
		if !v {
			return false
		}
	}
	return true
}
